baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/ConsistentInterfaces.java

```diff
@@ -0,0 +1,111 @@
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.MethodTreeMatcher;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.fixes.SuggestedFixes;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.VariableTree;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.code.Symbol.ClassSymbol;
+import com.sun.tools.javac.code.Symbol.MethodSymbol;
+import com.sun.tools.javac.code.Types;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "ConsistentInterfaces",
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        providesFix = BugPattern.ProvidesFix.REQUIRES_HUMAN_ATTENTION,
+        severity = SeverityLevel.ERROR,
+        summary = "Sub-types should have variable names consistent with the super-type")
+public final class ConsistentInterfaces extends BugChecker implements MethodTreeMatcher {
+    private static final Pattern UNKNOWN_ARG_NAME = Pattern.compile("^args\\d+$");
+
+    @Override
+    public Description matchMethod(MethodTree tree, VisitorState state) {
+        Symbol sym = ASTHelpers.getSymbol(tree);
+        if (!(sym instanceof MethodSymbol) || sym.isStatic()) {
+            return Description.NO_MATCH;
+        }
+
+        MethodSymbol methodSymbol = (MethodSymbol) sym;
+        return getNonParameterizedSuperMethod(methodSymbol, state.getTypes())
+                .filter(ConsistentInterfaces::hasMeaningfulArgNames)
+                .map(superMethod -> {
+                    SuggestedFix fix = IntStream.range(0, superMethod.params.size())
+                            .mapToObj(i -> {
+                                String param = methodSymbol.params.get(i).name.toString();
+                                String superParam =
+                                        superMethod.params.get(i).name.toString();
+                                if (param.equals(superParam)) {
+                                    return Stream.<SuggestedFix>empty();
+                                }
+                                VariableTree paramTree = tree.getParameters().get(i);
+                                return Stream.of(SuggestedFixes.renameVariable(paramTree, superParam, state));
+                            })
+                            .flatMap(Function.identity())
+                            .reduce(SuggestedFix.builder(), SuggestedFix.Builder::merge, SuggestedFix.Builder::merge)
```

Comment:
I think this would be easier to follow if we used `state.reportMatch` on each parameter rather the complex stream-based implementation. See the `BracesRequired` check as an example.